main.go

```diff
@@ -44,6 +48,27 @@ func createApp() *cli.App {
 	return app
 }
 
+func addBefore(app *cli.App, debug *bool) {
+	app.Before = func(c *cli.Context) error {
+		if *debug {
```

Comment:
`*debug` can be replaced by `c.Bool("debug")`
this way we avoid the usage of a global variable, instead we take out the value of the flag from the context

Reply:
I agree. We could use the same approach in the actions too. Will correct it for debug on this branch.